### ISSUE TITLE
Hunter‘s pets will fall underground

### ISSUE THREAD

**Issue by aleigood** [TrinityCore/TrinityCore#24267]
Description:
When you walk down a steep hill with your pet, the pet will go underground and sometimes disappear completely.
![bug4](https://user-images.githubusercontent.com/3367999/76146586-ec2b1f80-60ce-11ea-898e-b004a3a944f1.gif)

Branch(es):
3.3.5

TC rev. hash/commit:
961adea

Operating system:
CentOS

**Comment by illfated** [comment]
True. I have seen that happen so often that I have stopped caring about it. To me, now, this is only a cosmetic issue, but I do not know if the cause is related to mmaps, vmaps, or something else.

**Comment by aleigood** [comment]
> 真正。我看到这种情况经常发生，以至于我不再关心它。对我而言，这只是一个表面问题，但我不知道原因是否与mmap，vmap或其他有关。

**Comment by aleigood** [comment]
> 真正。我看到这种情况经常发生，以至于我不再关心它。对我而言，这只是一个表面问题，但我不知道原因是否与mmap，vmap或其他有关。

**Comment by jackpoz** [comment]
any coordinates where that ca be tested ?

**Comment by illfated** [comment]
Anywhere where there is a more-than-player size bump to travel across, in rough terrain. If the player and pet lands on different ground levels, or if the player walks close to cliff walls, the pet will walk and/or stop inside the cliff walls.

**Comment by illfated** [comment]
I should probably mention that the gif in the OP comment is pretty accurate, the pet disappears into the cliff wall because it is not set to be at a 90° angle with the player (then again, it can always do that if the hunter's orientation forces it to go there).

**Comment by jackpoz** [comment]
got a .gps location anyway ? it saves time on finding a spot like that, so I can focus on fixing the issue

**Comment by illfated** [comment]
Fair enough. I will log in to the test server and play around a bit. I think it will be easy enough in low-level areas like Durotar. BBL.

**Comment by Aokromes** [comment]
the screenshot looks somewhere shadowmoon valley

**Comment by illfated** [comment]
Here are 2 relatively accessible repeatable spots in Kalimdor. For this to work, the hunter needs to "strafe" sideways with its pet closer to the hillside/slope.
Tested on the tcubuntu server with 2 orc hunter characters (level 17 female & level 74 male).
```
Agama'gor, Southern Barrens: (NE from Camp Taurajo)
----------------------------
You are outdoors.
Map: 1 (Kalimdor) Zone: 17 (The Barrens) Area: 1700 (Agama'gor) Phase: 1
X: -2008.199829 Y: -2207.242920 Z: 111.939774 Orientation: 0.230068
grid[28,27]cell[1,6] InstanceID: 0
 ZoneX: 47.666046 ZoneY: 53.590374
GroundZ: 111.939354 FloorZ: 111.939354 Have height data (Map: 1 VMap: 1 MMap: 1)

You are outdoors.
Map: 1 (Kalimdor) Zone: 17 (The Barrens) Area: 1700 (Agama'gor) Phase: 1
X: -2005.149414 Y: -2216.666748 Z: 98.342667 Orientation: 0.418564
grid[28,27]cell[1,6] InstanceID: 0
 ZoneX: 47.759045 ZoneY: 53.545227
GroundZ: 98.342468 FloorZ: 98.342468 Have height data (Map: 1 VMap: 1 MMap: 1)

XYZ: -2008.199829 -2207.242920 111.939774 (+ face 0° due north)

ZoneXY: 47.666046 53.590374


Spirit Rise, Thunder Bluff (near horde NPCs):
---------------------------------------------
You are outdoors.
Map: 1 (Kalimdor) Zone: 1638 (Thunder Bluff) Area: 1640 (Spirit Rise) Phase: 1
X: -1051.760376 Y: 250.028397 Z: 129.764877 Orientation: 4.350431
grid[30,32]cell[0,3] InstanceID: 0
 ZoneX: 25.546179 ZoneY: 28.995514
GroundZ: 129.766037 FloorZ: 129.766037 Have height data (Map: 1 VMap: 1 MMap: 1)

You are outdoors.
Map: 1 (Kalimdor) Zone: 1638 (Thunder Bluff) Area: 1640 (Spirit Rise) Phase: 1
X: -1058.085205 Y: 250.781906 Z: 123.049438 Orientation: 4.395200
grid[30,32]cell[0,3] InstanceID: 0
 ZoneX: 25.473984 ZoneY: 29.904470
GroundZ: 123.051590 FloorZ: 123.051590 Have height data (Map: 1 VMap: 1 MMap: 1)

XYZ: -1051.760376 250.028397 129.764877 (+ face 90° east)

ZoneXY: 25.473984 29.904470
```
The requirement for the pet dropping inside the texture to happen seems to be that the character needs to fall or have a very steep hillside close to player falling angle. The pet also needs to be closer to the ground side/wall side during the drop, as well as the landing needs to be such that the player character does not slide out or move sideways from the landing point when landing.

**Comment by illfated** [comment]
BTW, for the landing spot on Spirit Rise, it is also possible to walk towards it (and 1-2 steps past, to get the pet at the right spot) to have the pet go just under the texture.

**Comment by jackpoz** [comment]
could someone test it again on a recent revision ?

**Comment by aleigood** [comment]
It has been tested, and the problem still exists
You can take your pet down from this position:
.go xyz -3521.755127 2329.189941 83.357994 530

**Comment by illfated** [comment]
(tcubuntu.northeurope.cloudapp.azure.com)
[04:48:27] TrinityCore rev. d81a9e5bc3b3 2020-06-03 22:05:08 -0300 (3.3.5 branch) (Unix, RelWithDebInfo, Static)
[04:48:55] Changed Channel: [1. General - Shadowmoon Valley]
[04:48:55] Changed Channel: [3. LocalDefense - Shadowmoon Valley]

Confirmed, starting from the provided command `.go xyz -3521.755127 2329.189941 83.357994 530` you just make the hunter (or other class with pet) walk straight forward until you drop down. Since the hunter pet is slightly behind and to your left, it will be hidden inside the hillside. (BTW, this looks like the same location as in the image in the main ticket comment.)
